### PR TITLE
Add McMMOPlayerSalvageCheckEvent

### PR DESCRIPTION
**Pull request by joblo2213**
[BetonQuest](https://github.com/Co0sh/BetonQuest) currently has compatibillity issues with mcMMOs salvage skill: 

Players can salvage items from quests which breaks those quests and can also be abused in many ways.

To fix this I need this event which allows me to cancel salvaging of quest items.

### CODE REVIEW

**Comment by joblo2213** [comment]
Any update on this?

File: src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+     * @return The enchanted book that should drop after salvaging or null if no book should be dropped.
+     */
+    public ItemStack getEnchantedBook() {
+        return enchantedBook;
+    }
+
+    /** Following are required for Cancellable **/
+    @Override
+    public boolean isCancelled() {
+        return false;
```
Comment (t00thpick1) [inline src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java]: If this always returns false setCancelled is pretty dang useless....

Comment (joblo2213) [reply src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java]: So first of all thanks for taking the time to have a look at this and doing a review :+1:
Looks like I was to much in a rush for getting this finished and actually forgot to propperly implement getCancelled() and setCancelled() which my IDE created for me. :face_palm:
So sorry for this inconvinience.

**Comment by joblo2213** [comment]
@t00thpick1 Fixed the issue and squashed the commits to one to keep everything clear.
Hope this can be merged now^^

File: src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        this.salvageMaterial = salvageMaterial;
+        this.salvageResults = salvageResults;
+        this.enchantedBook = enchantedBook;
+        this.cancelled = false;
+    }
+
+    /**
+     * @return The amount of materials you get back after salvaging.
+     */
+    public int getSalvageableAmount() {
```
[the remark this answers is not in the document]
Comment (joblo2213) [reply src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java]: Yea you are right, salvageItem fits way better than salvage material. 
Renamed it and overwrote the commit.

File: src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        return salvageableAmount;
+    }
+
+    /**
+     * @return The item that should get salvaged.
+     */
+    public ItemStack getSalvageMaterial() {
```
[the remark this answers is not in the document]
Comment (joblo2213) [reply src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java]: Yea you are right, salvage item fits way better than salvage material. 
Renamed it and overwrote the commit.

File: src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (joblo2213) [reply src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java]: Yeah you are right. 
I though this was easier to use that way but its more ore less a duplicate. 
I can remove it if you think its better without.

[the remark this answers is not in the document]
Comment (t00thpick1) [reply src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java]: Remove amount and I can pull it in.

[the remark this answers is not in the document]
Comment (joblo2213) [reply src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java]: ~Done

**Pull request merged**